scripts/perftests/PerformanceTestUtilities.ps1

```diff
@@ -0,0 +1,214 @@
+        return $Path;
+    }
+
+    # Writes the content to the given path. Creates the folder structure if needed
+    function OutFileWithCreateFolders([string]$path, [string]$content){
+        $folder = [System.IO.Path]::GetDirectoryName($path)
+        If(!(Test-Path $folder))
+        {
+            & New-Item -ItemType Directory -Force -Path $folder > $null
+        }
+        Add-Content -Path $path -Value $content
+    }
+
+    # Gets a list of all the nupkgs recursively in the global packages folder
+    function GetAllPackagesInGlobalPackagesFolder([string]$packagesFolder)
+    {
+        if(Test-Path $packagesFolder)
+        {
+            $packages = Get-ChildItem $packagesFolder\*.nupkg -Recurse
+            return $packages
+        }
+        return $null
+    }
+
+    # Gets a list of all the files resursively in the given folder
+    function GetFiles([string]$folder)
+    {
+        if(Test-Path $folder)
+        {
+            $files = Get-ChildItem $folder -recurse
+            return $files
+        }
+        return $null
+    }
+
+    # Determines if the client is dotnet.exe by checking the path.
+    function GetClientName([string]$nugetClient)
+    {
+        return $nugetClient.Substring($($nugetClient.LastIndexOf([System.IO.Path]::DirectorySeparatorChar) + 1))
+    }
+
+    function IsClientDotnetExe([string]$nugetClient)
+    {
+        return $nugetClient.EndsWith("dotnet.exe")
+    }
+
+    # Downloads the repository at the given path.
+    function DownloadRepository([string]$repository, [string]$commitHash, [string]$sourceDirectoryPath)
+    {
+        if(!(Test-Path $sourceDirectoryPath))
+        {
+                git clone $repository $sourceDirectoryPath
+                git -c $sourceDirectoryPath checkout $commitHash
+        }
+        else 
+        {
+                Log "Skipping the cloning of $repository as $sourceDirectoryPath is not empty" -color "Yellow"
+        }
+    }
+    
+    # Find the appropriate solution file for the repository. Looks for a solution file matching the repo name, 
+    # if not it takes the first available sln file in the repo. 
+    function GetSolutionFile([string]$repository,[string]$sourceDirectoryPath) {
+
+        $gitRepoName = $repository.Substring($($repository.LastIndexOf('/') + 1))
+        $potentialSolutionFile = [System.IO.Path]::Combine($sourceDirectoryPath, "$($gitRepoName.Substring(0, $gitRepoName.Length - 4)).sln")
+
+        if(Test-Path $potentialSolutionFile)
+        {
+            $solutionFile = $potentialSolutionFile
+        } 
+        else 
+        {
+            $possibleSln = Get-ChildItem $sourceDirectoryPath *.sln
+            if($possibleSln.Length -eq 0)
+            {
+                Log "No solution files found in $sourceDirectoryPath" "red"
+            } 
+            else 
+            {
+            $solutionFile = $possibleSln[0] | Select-Object -f 1 | Select-Object -ExpandProperty FullName
```

Comment:
I'd want to describe a test repository with:

* download URL
* the path to the solution to build

Taking the first solution found seems unpredictable.

Reply:
I agree that'd be better, but at this point there are no test cases that do that. 
We can simply extend it when needed.